tapiro/core/src/main/scala/io/buildo/tapiro/Meta.scala

```diff
@@ -7,17 +7,19 @@ import scala.meta._
 import cats.data.NonEmptyList
 
 object Meta {
-  val codecsImplicits = (routes: List[TapiroRoute]) => routes.flatMap {
+  val codecsImplicits = (routes: List[TapiroRoute]) =>
+    routes.flatMap {
       case TapiroRoute(route, errorValues) =>
-        errorValues.map(_.name) ++
-          route.body.map(b => typeNameString(b.tpe)) :+
-          typeNameString(route.returns)
-    }.distinct.map(stringToImplicitParam)
-
-  private[this] val stringToImplicitParam = (name: String) => {
-    val paramName = Term.Name(s"${name.head.toLower}${name.tail}")
-    val nameType = Type.Name(name)
-    param"implicit ${paramName}: JsonCodec[$nameType]"
+        errorValues.map(m => MetarpheusType.Name(m.name)) ++
```

Comment:
Here we only have the type name and so I'm not using `toScalametaType`. In any case, currently metarpheus doesn't support parametric types as members of tagged unions, so it shouldn't make any difference.